templates/template.rb

```diff
@@ -417,6 +419,34 @@ def initialize(config, flags)
             # changed to use fetch instead of delete.
             comment = options.delete(:comment)
 
+            if kinds = options[:kind]
+              kinds = [kinds] unless kinds.is_a?(Array)
+              kinds = kinds.map do |kind|
+                case kind
+                when "non-void expression"
+                  # the actual list of types would be way too long
+                  "Node"
+                when "pattern expression"
+                  # the list of all possible types is too long with 37+ different classes
+                  "Node"
+                when Hash
+                  kind = kind.fetch("on error")
+                  REMOVE_ON_ERROR_TYPES ? nil : kind
+                else
+                  kind
+                end
+              end.compact
+              if kinds.size == 1
+                kinds = kinds.first
+                kinds = nil if kinds == "Node"
```

Comment:
I initially tried for the kind to just be `"Node"` if there is nothing more precise but that causes a bunch of subtle bugs in templates, so I make it `nil` in that case like before. It's necessary to treat `Node` specially in several cases and comparing against a String doesn't seem any better.